core/src/main/scala/org/apache/spark/shuffle/sort/SortShuffleWriter.scala

```diff
@@ -72,6 +72,18 @@ private[spark] class SortShuffleWriter[K, V, C](
       val partitionLengths = sorter.writePartitionedFile(blockId, tmp)
       shuffleBlockResolver.writeIndexFileAndCommit(dep.shuffleId, mapId, partitionLengths, tmp)
       mapStatus = MapStatus(blockManager.shuffleServerId, partitionLengths)
+      partitionLengths.foreach(writeMetrics.incBlockSizeDistribution(_))
+      if (mapStatus.isInstanceOf[HighlyCompressedMapStatus]) {
+        writeMetrics.setAverageBlockSize(
+          mapStatus.asInstanceOf[HighlyCompressedMapStatus].getAvgSize());
```

Comment:
in scala, `if( isInstanceOf) { asInstanceOf}` can be replaced by pattern matching:

```scala
mapStatus match {
  case hc: HighlyCompressedMapStatus => writeMetrics.setAverageBlockSize(hc.getAvgSize());
  case _ =>  //no-op
}
```